clusterloader2/README.md

```diff
@@ -1,21 +1,77 @@
 
 To run ClusterLoader type:
 ```
-go run cmd/clusterloader.go --kubeconfig=kubeConfig.yaml --testconfig=config.json
+go run cmd/clusterloader.go --kubeconfig=kubeConfig.yaml --testconfig=config.yaml
 ```
 Flags kubeconfig and testconfig are necessary.
 
-#### Flags
+### Flags
 
+#### Required
+
+These flags are required for any test to be run.
  - kubeconfig - path to the kubeconfig file.
- - testconfig - path to the test config file.
+ - testconfig - path to the test config file. This flag can be used multiple times
+if more than one test should be run.
 
-### Vendor
+#### Optional
 
-Vendor is created using [govendor].
+ - nodes - number of nodes in the cluster.
+If not provided, test will assign the number of schedulable cluster nodes.
+ - report-dir - path to directory, where summaries files should be stored.
+If not specified, summaries are printed to standard log.
+
+## Tests
+
+### Test definition
+
+Test definition is an instantiation of this [api] (in json or yaml).
+The motivation and description of the API can be find in [design doc].
+Definitions of test as well as definitions of individual objects support templating.
+Templates for test definition come with one predefined value - ```{{.Nodes}}```, which represents the number of schedulable nodes in the cluster. \
+Example of a test definition can be found here: [load test].
+
+### Object template
 
+Object template is similar to standard kubernetes object definition with the only difference being templating mechanism.
+Parameters can be passed from the test definition to the object template using the ```templateFillMap``` map.
+Two always available parameters are ```{{.Name}}``` and ```{{.Index}}``` which specifies object name and object replica index respectively. \
+Example of a template can be found here: [load rc template].
+
+## Measurement
+
+Currently available measurements are:
+- **APIResponsiveness** \
+This metric creates summary for latency and number for server api calls. Api calls are divided by resource, subresource, verb and scope.
+- **CPUProfile** \
+This metric gathers the cpu usage profile provided by pprof for a given component.
+- **EtcdMetrics** \
+EtcdMetrics creates summary for etcd operations durations and etcd database size. The durations are represented as histograms.
+- **MemoryProfile** \
+This metric gathers the memory profile provided by pprof for a given component.
+- **MetricsForE2E** \
+The metric gathers e2e metrics from kube-apiserver, controller manager, all kubelets (optional based on metric parameter) and scheduler.
+- **SchedulingMetrics** \
+SchedulingMetrics summarizes the scheduler latency metrics for predicate, priority and preemption evaluation and binding.
+- **Timer** \
+Timer allows to create multiple instances of a timer for measuring time.
+Each timer is started and stopped using the measurement call defined in the test definition. After calling "gather" action, the Timer results will be logged into a file.
```

Comment:
This is not needed:
- the fast that you need to put it into test definition is obvious - it's the case for all measurements.
- the fact about logging is also common for all measurements